bson/codec_options.py

```diff
@@ -293,7 +289,7 @@ class CodecOptions(_options_base):
 
     def __new__(
         cls: Type["CodecOptions"],
-        document_class: Union[Type[MutableMapping], Type["RawBSONDocument"]] = dict,
+        document_class: Union[Type[MutableMapping], Type[Mapping], Type[Dict]] = dict,
```

Comment:
It's not clear to me how this change will benefit users. Can you explain?

Reply:
It was to match the types used in `_DocumentOut` for symmetry.

Reply:
Okay then I have the same question for _DocumentOut.

Reply:
I simplified all of the document types to be either `MutableMapping[str, Any]` or `RawBSONDocument`.